assets/src/edit-story/components/movable-element/index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import Moveable from 'react-moveable';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import { getComponentForType } from '../../elements';
+
+const Element = styled.div`
+	cursor: pointer;
+	user-select: none;
+`;
+
+const MovableElement = ( props ) => {
+	const {
+		rotationAngle,
+		x,
+		y,
+		type,
+		selected,
+		id,
+		rest,
+	} = props;
+
+	const [ targetEl, setTargetEl ] = useState( false );
+
+	const moveable = useRef();
+
+	const {
+		actions: { setPropertiesOnSelectedElements, setPropertiesById, selectElementById, toggleElementIdInSelection },
+	} = useStory();
+
+	const frame = {
+		translate: [ 0, 0 ],
+		rotate: rotationAngle,
+	};
+
+	const setStyle = ( target ) => {
+		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
+	};
+
+	const resetMoveable = ( target ) => {
+		frame.translate = [ 0, 0 ];
+		setStyle( target );
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	};
+
+	const handleSelectElement = useCallback( ( elId, evt ) => {
+		if ( evt.metaKey ) {
+			toggleElementIdInSelection( elId );
+		} else {
+			selectElementById( elId );
+		}
+		evt.stopPropagation();
+	}, [ toggleElementIdInSelection, selectElementById ] );
+
+	const comp = getComponentForType( type );
+	const Comp = comp; // why u do dis, eslint?
+
+	return (
+		<>
+			<Element
+				key={ id }
+				onClick={ ( evt ) => handleSelectElement( id, evt ) }
+			>
+				<Comp { ...rest } forwardedRef={ setTargetEl } />
+			</Element>
+			{ targetEl && <Moveable
+				ref={ moveable }
+				target={ targetEl }
+				draggable={ true }
+				resizable={ selected }
```

Comment:
Would be happy to be convinced otherwise or dig into this deeper. But I think your previous model was better: single moveable can support a single element, or a group of elements, or any additional customizations. Perhaps it's not an issue with moveable as much as we need to switch to do selection with `onPointerDown`? I looked around in other products (such as Google Slides) and it looks like they all do `onPointerDown` selection. That would solve our initial issue, right?

Reply:
It would solve the issue for draggable, however, should it be possible to resize/rotate the element before it is selected first? E.g. displaying the handles when hovering close to the element, without selecting (while keeping the same for the selected element, too). Perhaps only dragging should be possible without selecting first, that's how it works in Google Slides and Keynote, too. Will confirm just in case with the UX but I guess you're right and only dragging can be done without selecting first, which could be addressed with using `onPointerDown` for selecting.

Reply:
Hmm, actually, I wonder if `onPointerDown` would work since if `Moveable` would be added only once selected, then a new drag event should be initiated for the newly added `Moveable` to take effect.

Reply:
@dvoytenko Perhaps I'm missing something but it's not working currently for me -- setting the moveable target with `onPointerDown` (selecting the element which triggers the target setting) without having anything selected previously, it needs a new dragevent. I'll continue looking into this tomorrow morning, let me know if you had a working test somewhere, checked it briefly only at this moment.